Sources/EdgeState.swift

```diff
@@ -74,4 +90,57 @@ class EdgeState {
         currentCollectConsent = status
         hitQueue.handleCollectConsentChange(status: status)
     }
+
+    /// Get the current Edge Network location hint. May return `nil` if the hint has expired or is not set.
+    /// - Returns: the Edge Network location hint or nil if expired or not set
+    func getLocationHint() -> String? {
+        queue.sync {
+            return edgeProperties.locationHint
+        }
+    }
+
+    /// Update the Edge Network location hint and persist the new hint to the data store. If the new location hint is different from the previous, then a shared state
+    /// is also created with the new location hint value.
+    /// - Parameters:
+    ///   - hint: the Edge Network location hint to set
+    ///   - ttlSeconds: the time-to-live in seconds for the given location hint
+    ///   - createSharedState: function which creates a shared state with the Event Hub
+    func setLocationHint(hint: String, ttlSeconds: TimeInterval, createSharedState: @escaping (_ data: [String: Any], _ event: Event?) -> Void) {
+        queue.async {
+            let needsStateUpdate = self.edgeProperties.setLocationHint(hint: hint, ttlSeconds: ttlSeconds)
+            self.edgeProperties.saveToPersistence()
+
+            if needsStateUpdate {
+                // Create shared state if location hint changed
+                // Important - Using nil Event here which creates a shared state at the next available Event number.
+                //             An extension should NOT mix creating shared states using nil and using received events
+                //             as it can cause shared state generation to fail due to received events having potentially
+                //             lower event numbers than states using nil. If this extension later needs to create shared
+                //             states from received events, then this code must be refactored to also use received
+                //             events as the state version.
+                createSharedState(self.edgeProperties.toEventData(), nil)
```

Comment:
I'm using `nil` here for the shared state version, which will cause the Event Hub to get the next available version number. I'm using nil because this function is called by the NetworkResponseHandler which does not have a triggering event to create a shared state with. Using nil here means I must always use nil when creating a shared state even when I have one available (such as in bootupIfNeeded). The shared state is created mainly for Rules use-cases where a customer may want to build a postback URL using the location hint.

Another option is for NetworkResponseHandler to dispatch an event which is listened for by Edge to handle the location hint. That way there is an event to use to create the shared state. This is how Identity Direct extension handles this case. This option, however, will cause a delay when the location hint is handled and any queued hits will not include the hint in their URLs.

 As there are not current plans for Edge to include other properties in it's shared state, I feel the first option of using 'nil' is the better approach, but I wanted to point this out to hear other's thoughts.